robottelo/ui/contentsearch.py

```diff
@@ -0,0 +1,146 @@
+
+    def add_filter(self, filter_type, filter_value, auto_complete=None):
+        """Add necessary amount of filters ('autocomplete' fields)
+        At that moment we support next filters:
+        'Content Views'
+        'Products'
+        'Repositories' (second radiobutton)
+
+        Specify exact string that you want to be chosen from suggestions box in
+        auto_complete parameter. For example, you input 'MyRepo' into field and
+        see 'MyRepo01' and 'MyRepo02' in suggestion box, so it is necessary to
+        pass corresponding value into mentioned parameter (like auto_complete=
+        'MyRepo01')
+
+        """
+        self.select_filter_type(filter_type)
+        if filter_type == 'Repositories':
+            self.wait_until_element(
+                locators['contentsearch.repositories_auto_radio']
+            ).click()
+        self.field_update(
+            'contentsearch.{}'.format(self.ft_locator),
+            filter_value
+        )
+        strategy = locators['contentsearch.autocomplete_field'][0]
+        value = locators['contentsearch.autocomplete_field'][1]
+        self.wait_until_element((strategy, value % filter_value))
+        if auto_complete is not None:
+            self.wait_until_element((strategy, value % auto_complete)).click()
+        self.wait_for_ajax()
+        add_button = self.wait_until_element_is_clickable(
+            locators['contentsearch.add_{}_filter'.format(self.ft_locator)]
+        )
+        if add_button is None:
+            raise UIError(
+                'There is no "{0}" entity in the system'.format(filter_value)
+            )
+        add_button.click()
+
+    def add_search_criteria(self, filter_type, filter_value):
+        """Add search criteria (normal search fields)
+        At that moment we support next criterias:
+        'Repositories' (first radiobutton)
+        'Packages'
+        'Puppet Modules'
+
+        """
+        self.select_filter_type(filter_type)
+        if filter_type == 'Repositories':
+            self.wait_until_element(
+                locators['contentsearch.repositories_search_radio']
+            ).click()
+        self.field_update(
+            'contentsearch.{}_search'.format(self.ft_locator),
+            filter_value
+        )
+
+    def search(self, expected_result_list, result_view=None):
+        """Start searching procedure according to provided information
+        It is necessary to pass expected result to that function. For example,
+        you have next result on your page in browser:
+        TestContentView_01
+        --TestProduct01
+        ----TestRepository01
+        ------TestPackage01
+
+        In that case, you should have next parameter:
+        [
+        ['Content View', 'TestContentView_01', True],
+        ['Product', 'TestProduct01', True],
+        ['Repository', 'TestRepository01', True],
+        ['Package', 'TestPackage01', False]
+        ]
```

Comment:
You can denote a code block like so:

    In that case, you should have next parameter::

        [
            ['Content View', 'TestContentView_01', True],
            ['Product', 'TestProduct01', True],
            ['Repository', 'TestRepository01', True],
            ['Package', 'TestPackage01', False]
        ]